### PR TITLE
docs/resource/aws_eks_node_group: Add ignore_changes example, similar to aws_ecs_service resource

### LINKED ISSUE #11296: EKS Managed node groups autoscaling
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In the aws_eks_node_group scaling_config block you are required to set the desired size. This is undesirable b/c I want to let my nodes autoscale within the min-max range I set. Everytime my terraform is deployed it tries to reset the ASG to the desired size set within terraform.

I believe desired size should be optional. When you use an ASG directly with Terraform the desired_capacity is optional and would expect the same for this API that manages an ASG behind the scenes.

### New or Affected Resource(s)

Resource: aws_eks_node_group
Block: scaling_config

### Potential Terraform Configuration

```hcl
  scaling_config {
    max_size     = 1
    min_size     = 1
  }
```

### References

* https://www.terraform.io/docs/providers/aws/r/eks_node_group.html

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #11296
Reference: https://registry.terraform.io/providers/hashicorp/aws/2.62.0/docs/resources/ecs_service#ignoring-changes-to-desired-count

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing: N/A (documentation)

### CODE REVIEW

File: website/docs/r/eks_node_group.html.markdown [hunk 1]
```diff
@@ -35,6 +35,28 @@ resource "aws_eks_node_group" "example" {
 }
 ```
 
+### Ignoring Changes to Desired Size
+
+You can utilize the generic Terraform resource [lifecycle configuration block](/docs/configuration/resources.html#lifecycle) with `ignore_changes` to create an EKS Node Group with an initial size of running instances, then ignore any changes to that count caused externally (e.g. Application Autoscaling).
```
Comment (anGie44) [inline website/docs/r/eks_node_group.html.markdown:40]: will the docs file be added to this repo? wasn't sure if there's an upcoming change there or if we want to refer to the external lifcycle link: https://www.terraform.io/docs/configuration/resources.html?&_ga=2.203417043.394886334.1589905661-628874299.1583248857#lifecycle-lifecycle-customizations

Comment (bflad) [reply website/docs/r/eks_node_group.html.markdown:40]: The general terraform.io/docs files live in https://github.com/hashicorp/terraform-website and providers will not be expected to maintain them separately at any point. When the provider documentation is officially migrated to the Terraform Registry, then we will change all these absolute path links with fully absolute links (e.g. https://terraform.io/docs/...), but this is waiting to be done all at once as far as I know. I will update the anchor so its more accurate though.

Comment (bflad) [reply website/docs/r/eks_node_group.html.markdown]: (For more context: absolute path links to /docs/... in the Registry documentation are automatically rewritten to point at https://terraform.io/docs/... to not make this a problem during the transition period -- not sure if this is done during serving the documentation or during provider ingress off the top of my head)

Comment (anGie44) [reply website/docs/r/eks_node_group.html.markdown]: ah no wonder, cool make sense!

**Pull request merged**